template/data/src/main/java/co/nimblehq/template/data/extensions/ResponseMapping.kt

```diff
@@ -0,0 +1,53 @@
+package co.nimblehq.template.data.extensions
+
+import co.nimblehq.template.data.response.ErrorResponse
+import co.nimblehq.template.data.service.ApiException
+import co.nimblehq.template.data.service.NoConnectivityException
+import co.nimblehq.template.data.service.providers.MoshiBuilderProvider
+import com.squareup.moshi.JsonDataException
+import kotlinx.coroutines.flow.FlowCollector
+import kotlinx.coroutines.flow.flow
+import retrofit2.HttpException
+import java.io.IOException
+import java.io.InterruptedIOException
+import java.net.UnknownHostException
+import kotlin.experimental.ExperimentalTypeInference
+
+@Suppress("TooGenericExceptionCaught")
+@OptIn(ExperimentalTypeInference::class)
+fun <T> flowTransform(@BuilderInference block: suspend FlowCollector<T>.() -> T) = flow {
```

Comment:
Is the `@BuilderInference` annotation really necessary here? The method appears to work even without it 🤔 Not sure if I've missed anything 😄

Reply:
Ah, so we don't have to declare the type explicitly 👍🏻 
So the reason this worked on my end is because it was still declared as 
`override fun getModels(): Flow<List<Model>> = flowTransform {` 😅

Reply:
@ryan-conway Updated the repository ✅

Reply:
@ryan-conway Java 1.8 does not support it yet. I reverted the Repository back to declare the type explicitly with `Flow<List<Model>>`. We can upgrade the Java to 11 and update this later.